functions/src/modules/SQR.functions.ts

```diff
@@ -303,118 +299,107 @@ export const processSubmission = functions.database
-    });
-
-    const getSubmissions = promisify(SubmissionsSheet.getRows);
-
-    // submissions = Submissions sheet rows
-    const submissions = await getSubmissions();
-    submissions.forEach(row => {
-      const serial = row['submissionserial'];
+    const spreadsheetId = functions.config().sqr.spreadsheet_id;
+
+    // Regex for validating file names in both sheets
+    const audioFileRegex = /(\w+)-\d+.*/;
+
+    ////////////////////////
+    //     Submissions
+    ////////////////////////
+    const submissionsSheet = new GoogleSheets(spreadsheetId, ISoundQualityReportSheet.Submissions);
+    const submissionsRows = await submissionsSheet.__getRows();
+    for (const row of submissionsRows) {
+      const audioFileName = row['Audio File Name'];
+      const match = audioFileRegex.exec(audioFileName);
+      if (!match) continue;
+
+      const list = match[1];
+      // const serial = row['Submission Serial'];
+
+      const fileRef = await db.ref(`/files/${list}/${audioFileName}/soundQualityReporting`).once('value');
+      const audioFile = fileRef.val();
+      let token;
+      if (fileRef.exists()) {
+        token = audioFile.token;
+      } else {
+        console.warn(`Token not found!`)
+        continue;
+      }
 
       const regex = /(.*?)–(.*):(.*)—(.*)/g;
-      const soundissuesMatch = regex.exec(row['soundissues']);
-      const unwantedpartsMatch = regex.exec(row['unwantedparts']);
+      const soundissuesMatch = regex.exec(row['Sound Issues']);
+      const unwantedpartsMatch = regex.exec(row['Unwanted Parts']);
 
-      const soundissues = {
-          beginning: soundissuesMatch[1],
-          ending: soundissuesMatch[2],
-          type: soundissuesMatch[3],
-          description: soundissuesMatch[4],
-        },
-        unwantedparts = {
-          beginning: unwantedpartsMatch[1],
-          ending: unwantedpartsMatch[2],
-          type: unwantedpartsMatch[3],
-          description: unwantedpartsMatch[4],
-        };
+      if (!soundissuesMatch || !unwantedpartsMatch) {
+        console.warn(`"Sound Issues" or "Unwanted Parts" are not in the correct format`);
+        continue;
+      }
 
+      const soundissues = {
+        beginning: soundissuesMatch[1],
+        ending: soundissuesMatch[2],
+        type: soundissuesMatch[3],
+        description: soundissuesMatch[4]
+      };
+      const unwantedparts = {
+        beginning: unwantedpartsMatch[1],
+        ending: unwantedpartsMatch[2],
+        type: unwantedpartsMatch[3],
+        description: unwantedpartsMatch[4]
+      };
       const submission = {
-        author: {
-          name: row['name'],
-          emailAddress: row['emailAddress'],
-        },
-        fileName: row['audiofilename'],
-        changed: row['changed'],
-        completed: row['completed'],
-        created: row['created'],
-        comments: row['comments'],
+        fileName: row['Audio File Name'],
```

Comment:
File name should not be saved in the submission anymore as it is specified in the path.